source/iface/unsampled-image-accessor.rst

```diff
@@ -2,20 +2,66 @@
   Copyright 2020 The Khronos Group Inc.
   SPDX-License-Identifier: CC-BY-4.0
 
-.. _unsampled-image-accessors:
+.. _unsampled_image_accessors:
 
 *************************
 Unsampled image accessors
 *************************
 
-.. rst-class:: api-class
+There are two classes which implement accessors for unsampled images,
+``unsampled_image_accessor`` and ``host_unsampled_image_accessor``.
+The former provides access from within a SYCL kernel function or from
+within a host task. The latter provides access from host
+code that is outside of a host task.
+
+The dimensionality of an unsampled image accessor must match the
+dimensionality of the underlying image to which it provides access.
+Both unsampled image accessor classes support the
+``access_mode::read`` and ``access_mode::write`` access modes.
+In addition, the ``host_unsampled_image_accessor``
+class supports ``access_mode::read_write``.
+
+The AccessTarget template parameter dictates how the
+``unsampled_image_accessor`` can be used: ``image_target::device``
+means the accessor can be used in a SYCL kernel function while
+``image_target::host_task`` means the accessor can be used in
+a host task. Programs which specify this template parameter as
+``image_target::device`` and then use the ``unsampled_image_accessor``
+from a host task are ill formed. Likewise, programs which specify this
+template parameter as ``image_target::host_task`` and then use the
+``unsampled_image_accessor`` from a SYCL kernel function are ill formed.
 
 .. _unsampled_image_accessor:
 
 ==================================
 ``sycl::unsampled_image_accessor``
 ==================================
 
+::
+
+  class unsampled_image_accessor;
```

Comment:
This is missing the template parameters, which makes it hard to understand the text above.  Please add the full declaration including template parameters here.

Reply:
Does it match your idea with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/103/commits/3b7703b9b546c60fc54f01f0d13297f08a2769bc

Reply:
I was thinking more of the snip just saying:
```
template <typename DataT, int Dimensions, access_mode AccessMode,
          image_target AccessTarget = image_target::device>
class unsampled_image_accessor
```

Later parts of the page talk about the template parameters being used to control which interfaces are available, and without showing the full parameterization, it isn't clear where things like AccessMode come from.

Reply:
Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/103/commits/c0ddbd63cf04bf73fb8e85357dcf5aecf0e4805a